### PR TITLE
cobbler inventory plugin: Convert unicode to str

### LINKED ISSUE #6922: Cobbler inventory plugin auth
### Summary

Looks like when logging into cobbler, plugin passes `yaml.objects.AnsibleUnicode` to `xmlrpc/client.py` and it doesn't understand what to do with it.

Found out this by printing type of object that is returned by `self.get_option('user')` or `self.get_option('password')`
<class 'ansible.parsing.yaml.objects.AnsibleUnicode'>

If i would wrap it in `str(self.get_option('user'))` it works as expected.

Is it intentional? Maybe it has something to do with specific ansible version? Thanks.

### Issue Type

Bug Report

### Component Name

plugins.inventory.cobbler

### Ansible Version

```console (paste below)
$ ansible --version
ansible [core 2.11.12]
```

### Community.general Version

```console (paste below)
$ ansible-galaxy collection list community.general
Collection        Version
----------------- -------
community.general 7.1.0 
```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed

```

### OS / Environment

_No response_

### Steps to Reproduce

```yaml (paste below)

```

### Expected Results

It should login to cobbler with correct password

### Actual Results

```console (paste below)
failis with `raise TypeError("cannot marshal %s objects" % type(value))` from xmlrpc client
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by Algirdas-Z**
##### SUMMARY

Explicitly convert AnsibleUnicode to str() before passing it to xmlrpc client

Fixes #6922 

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

community.general.cobbler

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/inventory/cobbler.py [hunk 1]
```diff
@@ -145,7 +145,7 @@ def _get_connection(self):
             self.connection = xmlrpc_client.Server(self.cobbler_url, allow_none=True)
             self.token = None
             if self.get_option('user') is not None:
-                self.token = self.connection.login(self.get_option('user'), self.get_option('password'))
+                self.token = self.connection.login(str(self.get_option('user')), str(self.get_option('password')))
```
Comment (felixfontein) [inline plugins/inventory/cobbler.py:148]:
```suggestion
                self.token = self.connection.login(to_text(self.get_option('user')), to_text(self.get_option('password')))
```

File: changelogs/fragments/6923-cobbler-inventory_unicode.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - cobbler inventory plugin - apply ``str()`` method before passing user/password to xmlrpc client (https://github.com/ansible-collections/community.general/pull/6923).
```
Comment (felixfontein) [inline changelogs/fragments/6923-cobbler-inventory_unicode.yml:2]:
```suggestion
  - cobbler inventory plugin - convert Ansible unicode strings to native Python unicode strings before passing user/password to XMLRPC client (https://github.com/ansible-collections/community.general/pull/6923).
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution!

File: plugins/inventory/cobbler.py (hunk 1 above)
Comment (Algirdas-Z) [reply plugins/inventory/cobbler.py:148]: Please excuse me if my lack of knowledge is getting in a way, but shouldn't I see "string" type after "to\_text()" if that is what you do?  
just added plain print to that plugin to verify:  
`print("str object type:{}, to_text object type: {}".format(type(str(self.get_option('user'))), type(to_text(self.get_option('password')))))`

and I get 

`str object type:<class 'str'>, to_text object type: <class 'ansible.parsing.yaml.objects.AnsibleUnicode'>`  
so at the end, we have changed nothing, we are still forwading AnsibleUnicode to xmlrpc client.

Comment (felixfontein) [reply plugins/inventory/cobbler.py:148]: Hmm, in that case it seems that `AnsibleUnicode` is too clever for `to_text` to properly work. Please use
```
from ansible.module_utils.six import text_type

... text_type(get_option('user')) ... text_type(get_option('password')) ...
```
instead. That produces a real `str` on Python 3, and a real `unicode` on Python 2.

**Review by felixfontein — APPROVED** [review]
LGTM. I assume you tested this and it works for you?

**Comment by Algirdas-Z** [comment]
> LGTM. I assume you tested this and it works for you?

Yep, tested it on my setup, works fine ;)

**Comment by felixfontein** [comment]
If nobody objects, I'll merge this tomorrow.

**Comment by felixfontein** [comment]
@Algirdas-Z thanks for fixing this!

**Pull request merged**